### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Nike9486** [grafana/grafana#11235]
**What Grafana version are you using?**

v5.0.1 

**What datasource are you using?**

InfluxDB

**What OS are you running grafana on?**

RHEL 6

**What did you do?**

Trying to get rendered image using URL: [grafana host]/render/d/:uid/:slug

**What was the expected result?**

Got rendered image

**What happened instead?**

The page falling by timeout. In the log file we finded:

 t=2018-03-14T09:03:45+0300 lvl=info msg=Rendering logger=png-renderer path="d/000000112/[dashboard name]?from=now-1h&fullscreen=&height=500&orgId=15&panelId=68&timeout=120&to=now&tz=UTC%2B02%3A00&width=1000"
t=2018-03-14T09:03:46+0300 lvl=dbug msg="[phantom] Loading a web page: http://localhost:3000/d/000000112/[dashboard name]?from=now-1h&fullscreen=&height=500&orgId=15&panelId=68&timeout=120&to=now&tz=UTC%2B02%3A00&width=1000 status: success 120000" logger=png-renderer
t=2018-03-14T09:03:47+0300 lvl=dbug msg="[phantom] ERROR: Maximum call stack size exceeded.\nhttp://localhost:3000/public/build/vendor.2c3c591224fb193f7fd3.js:15\nTRACE:\n -> : 0 (in function \"apply\")" logger=png-renderer
t=2018-03-14T09:03:47+0300 lvl=dbug msg="[phantom] phantomjs://code/render.js:48 in onError" logger=png-renderer

**Comment by marefr** [comment]
So your trying to render a whole dashboard? otherwise use the `/render/d-solo/<uid>/<slug>` url. Do you use a reverse proxy?

**Comment by Nike9486** [comment]
No, we don't use a proxy and we just try to render one graph. We tryed to used /render/d-solo/<uid>/<slug> but nothing changes.

**Comment by marefr** [comment]
Have you tried changing the timeout to a higher value? Could you please include a screenshot of your panel when rendered in dashboard and possible also your panel json so I can try it out? Thanks

**Comment by Thrael** [comment]
attached.
[panel.png](https://user-images.githubusercontent.com/5838390/37597572-da60b794-2b90-11e8-9432-cdb41f13cffd.png)
[panel.txt](https://github.com/grafana/grafana/files/1825085/panel.txt)

**Comment by marefr** [comment]
Thanks. I tried your panel and can render it using phantomjs, however I don't have any data. I think that you should try and increase the timeout to a lot higher and see if that works better. I can see that you have a lot of influx queries why it make take a while to request and render all the series.

**Comment by Thrael** [comment]
Looks like problem reproduces only when you fold(hide) panel within dashboard.

**Comment by marefr** [comment]
@Thrael that should be fixed as soon as PR #11282 are accepted and merged

**Comment by cby016** [comment]
I'm having this issue with a graph that is not hidden. I'm running grafana 5.0.3 on mac and this is the error I'm getting.
```
t=2018-03-20T08:23:22-0600 lvl=dbug msg="Updating last user_seen_at" logger=context userId=1 orgId=1 uname= user_id=1
t=2018-03-20T08:23:24-0600 lvl=dbug msg="[phantom] Maximum call stack size exceeded.\nhttp://localhost:3000/public/build/vendor.23437b9523cd388481b5.js:15" logger=png-renderer
t=2018-03-20T08:23:24-0600 lvl=dbug msg="[phantom] :0 in apply" logger=png-renderer
t=2018-03-20T08:23:24-0600 lvl=dbug msg="[phantom] Loading a web page: http://localhost:3000/d-solo/c9fqwRkmk/energy?orgId=1&panelId=2&from=1521469137426&to=1521555537426&width=1000&height=500&tz=UTC-06%3A00&render=1 status: success 60000" logger=png-renderer
```

**Comment by daniellee** [comment]
@cby016 can confirm that Phantomjs and direct rendered links do not work with Grafana when installed with brew on macOS. Think that it is a separate issue from the original one. We will investigate more and open a new issue with details.

**Comment by ZachBenz** [comment]
@daniellee Can confirm experiencing the same issue under macOS with Grafana installed with brew.  Also interested in tracking the issue for this.

**Comment by daniellee** [comment]
@mlamoure @ZachBenz issue created #11638 and hopefully we can get it fixed before the 5.1 beta release tomorrow/Friday.

**Comment by torkelo** [comment]
try latest grafana v5.2 and this new render plugin, https://github.com/grafana/grafana-image-renderer  it uses headless chrome to render instead of old phantomjs
